aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -250,6 +264,24 @@
   @Deprecated
   public static final boolean CLIENT_ENABLE_ETAG_CHECK_DEFAULT = false;
 
+  /**
+   * Used by {@link LakeFormationAwsClientFactory}.
+   * The table name used as part of lake formation credentials request.
+   */
+  public static final String LAKE_FORMATION_TABLE_NAME = "lakeformation.table-name";
+
+  /**
+   * Used by {@link LakeFormationAwsClientFactory}.
+   * The database name used as part of lake formation credentials request.
+   */
+  public static final String LAKE_FORMATION_DB_NAME = "lakeformation.db-name";
+
+  /**
+   * Used by {@link LakeFormationAwsClientFactory}.
+   * The account ID used as part of lake formation credentials request.
+   */
+  public static final String LAKE_FORMATION_ACCOUNT_ID = "lakeformation.account-id";
```

Comment:
[question] Thinking out loud here : 
what if the end-user forget's to add any of these conf's, he will see AWS stacktrace that the ARN is invalid, could we in principle fail him earlier with IllegalArg / Validation Error, as we do by checking the awsClientFactory is instance of LF ?  WDYT

Reply:
Precondition null checks added, I think this is resolved.